### PR TITLE
ENH: _very broken mode_ for EllLinear

### PR DESCRIPTION
**Pull request by klauer**
* Fixes units for spectrum wavelength
* Adds "very broken mode" for EllLinear to better exercise retry mode in las-dispersion-scan
* Adds a fix for `write_if_changed` to better support numpy arrays

### CODE REVIEW

File: simioc/ioc/las_dscan.py [hunk 1]
```diff
@@ -103,18 +103,23 @@ def __init__(self, *args, dataset: Optional[Dataset] = None, **kwargs):
...
 
     @sim_enable.scan(period=0.2)
     async def sim_enable(self, instance: ChannelData, async_lib: AsyncLibraryLayer):
         if self.sim_enable.value not in (1, "On"):
             return
 
-        spectrum = self.dataset.find_closest_spectrum(
-            self.motor.user_readback_position,
-        )
+        pos = self.upstream.user_readback_position
```
Comment (slactjohnson) [inline simioc/ioc/las_dscan.py:114]: Why use just the upstream motor here, rather than the combined pseudo motor?

**Comment by klauer** [comment]
Bump @slactjohnson

**Review by slactjohnson — APPROVED** [review]
LGTM. I had one question, but this worked regardless.

File: simioc/ioc/las_dscan.py (hunk 1 above)
Comment (klauer) [reply simioc/ioc/las_dscan.py:114]: That's a good question - not sure what influenced that decision...

**Pull request merged**